src/core/ext/transport/chttp2/transport/internal.h

```diff
@@ -233,26 +232,40 @@ typedef enum {
   GRPC_CHTTP2_KEEPALIVE_STATE_DISABLED,
 } grpc_chttp2_keepalive_state;
 
-struct grpc_chttp2_transport : public grpc_core::KeepsGrpcInitialized {
+struct grpc_chttp2_transport final
+    : public grpc_core::Transport,
```

Comment:
It seems a little weird to inherit from both `Orphanable` (via `Transport`) and from `RefCounted<>`.  Instead, how about making `Transport` inherit from `InternallyRefCounted<>` (which inherits from `Orphanable`)?

Reply:
not all transports need that though, and refcounting need not be part of the transport api surface

(indeed it's likely we'll move to a small group of separately allocated objects within a given transport going forward)